src/common/lib/utils.ts

```diff
@@ -85,89 +86,101 @@ const utils = {
     const { top, left } = await getWindowPosition(windowWidth, windowHeight);
 
     return new Promise((resolve, reject) => {
-      browser.windows
-        .create({
-          url: url,
-          type: "popup",
-          width: windowWidth,
-          height: windowHeight,
-          top: top,
-          left: left,
-        })
-        .then((window) => {
-          let closeWindow = true; // by default we call remove.window (except the browser forces this prompt to open in a tab)
-          let tabId: number | undefined;
-          if (window.tabs) {
-            tabId = window.tabs[0].id;
-          }
+      const createPopup = (options: browser.Windows.CreateCreateDataType) => {
+        return browser.windows.create(options);
+      };
 
-          // Kiwi Browser opens the prompt in the same window (there are only tabs on mobile browsers)
-          // Find the currently active tab to validate messages
-          if (window.tabs && window.tabs?.length > 1) {
-            tabId = window.tabs?.find((x) => x.active)?.id;
-            closeWindow = false; // we'll only remove the tab and not the window further down
-          }
+      const createTab = (options: browser.Tabs.CreateCreatePropertiesType) => {
+        return browser.tabs.create(options);
+      };
 
-          // Re-focus the popup after 2 seconds to mitigate the problem of lost popups
-          // (e.g. when a user clicks the website)
-          setTimeout(() => {
-            if (!window.id) return;
+      const optionsPopup: browser.Windows.CreateCreateDataType = {
```

Comment:
why declare these outside of createTab/createPopup? if you put them inside then you don't need to pass arguments. Also, maybe it would be nice to suffix with "Options" or "Properties" since they aren't actually a popup/tab, only the config passed to create the popup or tab

Reply:
resolved